### DISCUSSION TITLE
How to disable the default basic authentication?

### DISCUSSION THREAD

**Discussion by JasonCaoJR** [internetarchive/heritrix3#584] [Q&A] [answered]
Hi,
The basic authentication stops all the REST API calls. There are always error to call the rest api, event with setting headers of:
--header 'Authorization: Basic YWRtaW46YWRtaW4='
How to disable the default basic authentication?

**Comment by ato** [comment, marked as answer]
Heritrix currently uses digest authentication not basic authentication. You'll need to use a client that supports digest authentication, for example using curl:

    curl -k --digest -u username:password https://localhost:8443/engine

Currently there are no options to change or disable the authentication. (The relevant code is [here](https://github.com/internetarchive/heritrix3/blob/a151937d4c9c0a9beceeb306161e66ea4b4a19f2/engine/src/main/java/org/archive/crawler/Heritrix.java#L358-L363).)
